### PR TITLE
feat(rust,python): add glob support to scan_ndjson

### LINKED ISSUE #6638: Dealing with multiple JSON files
### Problem description

The Polars book has a page about [scanning multiple CSV files into a single DataFrame](https://pola-rs.github.io/polars-book/user-guide/multiple_files/intro.html). 
I'd like to to the same by using `polars.scan_ndjson`, but it seems that this feature is not available when dealing with JSON files: Polars/Python returns an `OSError`: `OSError: No such file or directory (os error 2)`.

Is such a feature planned? It would be really useful when dealing with large sets of data split among several files.

Also, in the meantime, is there a way of easily replicating this behaviour? I feel like I could instantiate several `LazyFrame`, but then the merging would have to be done, right?

Thanks :)

### PR DESCRIPTION
**Pull request by mslapek**
Fixes #6638

### CODE REVIEW

File: examples/datasets/foods1.ndjson [hunk 1]
```diff
@@ -1,27 +0,0 @@
-{"category":"vegetables","calories":45,"fats_g":0.5,"sugars_g":2}
```
Comment (ritchie46) [inline examples/datasets/foods1.ndjson:1]: Was this intentional?

Comment (mslapek) [reply examples/datasets/foods1.ndjson:1]: Yes. The files `foods1.ndjson` and `foods2.ndjson` are created in polars/polars-lazy/src/tests/mod.rs

Comment (ritchie46) [reply examples/datasets/foods1.ndjson:1]: Ah, right. :+1:

**Comment by ritchie46** [comment]
Thanks @mslapek!

**Comment by mslapek** [comment]
@ritchie46 Thanks for the review! 🎉

**Pull request merged**